ls_colors.py

```diff
@@ -0,0 +1,240 @@
+        self.tup_ls_colors = []
+
+        # Not considering file extensions
+        # The order of these two block matters, as extensions colouring should
+        # take precedence over the 'file' type
+        for key in [k for k in self.ls_colors_keys if '.*' not in k]:
+            if key[0] == 'fi':
+                self.tup_ls_colors += [('file', key[1])]
+
+        # Considering files extensions
+        self.tup_ls_colors += [('.' + k[0].split('*.')[1], k[1])
+                               for k in self.ls_colors_keys if '*.' in k[0]]
+
+        # This is added last because their color should take precedence over
+        # what's been set before for a 'file' that would have the same 
+        # extension
+        for key in [k for k in self.ls_colors_keys if '.*' not in k]:
+            if key[0] == 'ex':
+                self.tup_ls_colors += [('executable', key[1])]
+            elif key[0] == 'pi':
+                self.tup_ls_colors += [('fifo', key[1])]
+            elif key[0] == 'ln':
+                self.tup_ls_colors += [('link', key[1])]
+            elif key[0] == 'bd' or key[0] == 'cd':
+                self.tup_ls_colors += [('device', key[1])]
+            elif key[0] == 'so':
+                self.tup_ls_colors += [('socket', key[1])]
+            elif key[0] == 'di':
+                self.tup_ls_colors += [('directory', key[1])]
+
+        # Those special context shouldn't get attributes destined to
+        # files, based on extension
+        self.__special_context = [
+            "directory",
+            "fifo",
+            "link",
+            "device",
+            "socket"
+        ]
+
+        self.progress_bar_color = 1
+
+    def new_hook_before_drawing(self, fsobject, color_list):
+        for key in self.ls_colors_extensions:
+            if fsobject.basename.endswith(key):
+                color_list.append(key)
+
+        return self.OLD_HOOK_BEFORE_DRAWING(fsobject, color_list)
+
+    def get_default_lscolors(self):
+        """Returns the default value for LS_COLORS
+        as parsed from the `dircolors` command
+        """
+        ls_colors = check_output('dircolors')
+        ls_colors = ls_colors.splitlines()[0].decode('UTF-8').split("'")[1]
+        return ls_colors
+
+    def get_attr_from_lscolors(self, attribute_list):
+        return_attr = 0
+        to_delete = []
+
+        for i, attr in enumerate(attribute_list):
+            if attr == 1:
+                return_attr |= style.bold
+            elif attr == 4:
+                return_attr |= style.underline
+            elif attr == 5:
+                return_attr |= style.blink
+            elif attr == 7:
+                return_attr |= style.reverse
+            elif attr == 8:
+                return_attr |= style.invisible
+            to_delete.append(i)
+
+        # remove style attrattributes  from the array
+        attribute_list[:] = [val for i, val in enumerate(attribute_list) if i in to_delete]
```

Comment:
Given that every `i` is added to `to_delete` unconditionally, this is essentially clearing the entire list. Is this intended?

Reply:
That does seem incorrect, yes. Though it is copying the entire list if I'm not mistaken.

Reply:
Yes, you're correct. It is copying. I saw `to_delete` and read that last condition as `not in`. I'm guessing that the intention was to filter out any of the style attributes that were matched in the loop above it.